### DISCUSSION TITLE
Why does blackhole audio surpass 0 dB when routed to Logic Pro?

### DISCUSSION THREAD

**Discussion by deddydrip** [ExistentialAudio/BlackHole#513] [General]
I've realized when using blackhole with Logic Pro, the volume peaks at 0 dB, but many times, the volume passes +1 dB and sometimes even up to +4 dB on the level meters. This even happens when running Apple Music through logic. The weird thing is that when I swap to my MacBook speakers, it is the same volume. When running through logic, I think it does cause distortion a little sometimes too, but maybe not, so even though the levels are passing 0 dB, I'm not sure if it is always clipping or not. 

This isn't really a problem, I am just a little curious about what causes this. is it caused by a synchronization issue, is it more of just a precision issue, or is it just as simple as all music played from Apple Music is not limited to the 0 dB limit that I thought it was?

I've looked everywhere, but haven't seen anyone else talking about this. maybe I'm not looking up the right thing.

Also just tried pan law setting they seemed to have done something, but I don't know if they fixed the problem or just solved it indirectly by changing something else, caus I didn't pan any of my tracks.

**Comment by devinroth** [comment]
That’s a good question.

I choose not to clip audio since BlackHole is using Float32 internally. The benefit is that audio won’t clip if it goes above 1.0 or below -1.0. I think it has a more desired effect than clipping.

I have no idea if Apple Music is manipulating the audio but in theory could boost beyond 1.0 if you have an eq or something.

At full volume BlackHole passes identical audio between output and input.

It’s also possible that Logic is a little trigger happy with the clipping meter.

Don’t forget that some music is mastered so loudly that the waveform looks like a sausage.

Devin

**Reply by deddydrip** [reply]
Thank you, for the quick reply! That makes sense a lot more sense. Honestly, it's probably a mix of all of these.

It's good to know that the audio is identical on both ends. 

I use BlackHole 24/7 and even built a second driver called "WhiteHole", which I use to output audio from logic to other apps, lol. 

Thank you for your work on this; I absolutely love the customizability of it! It was the first project I ever built on Xcode and got me interested in learning more about Digital Audio.

**Comment by devinroth** [comment]
Oh sweet. Glad I could be an inspiration.

Devin
